dpnp/dpnp_iface_histograms.py

```diff
@@ -208,6 +238,186 @@ def _get_bin_edges(a, bins, range, usm_type):
+    min_v = dpnp.min(x_casted)
+
+    if min_v < 0:
+        raise ValueError("x argument must have no negative arguments")
+
+    size = int(dpnp.max(max_v)) + 1
+    if minlength is not None:
+        size = max(size, minlength)
+
+    # bincount implementation uses atomics, but atomics doesn't work with
+    # host usm memory
+    n_usm_type = "device" if usm_type == "host" else usm_type
+
+    # bincount implementation requires output array to be filled with zeros
+    n_casted = dpnp.zeros(
+        size, dtype=n_dtype, usm_type=n_usm_type, sycl_queue=queue
+    )
+
+    _manager = dpu.SequentialOrderManager[queue]
+
+    x_usm = dpnp.get_usm_ndarray(x_casted)
+    weights_usm = (
+        dpnp.get_usm_ndarray(weights_casted)
+        if weights_casted is not None
+        else None
+    )
+    n_usm = dpnp.get_usm_ndarray(n_casted)
+
+    mem_ev, bc_ev = statistics_ext.bincount(
+        x_usm,
+        min_v,
+        max_v,
+        weights_usm,
+        n_usm,
+        depends=_manager.submitted_events,
+    )
+
+    _manager.add_event_pair(mem_ev, bc_ev)
+
+    return n_casted
+
+
+def bincount(x, weights=None, minlength=None):
+    """
+    bincount(x, /, weights=None, minlength=None)
+
+    Count number of occurrences of each value in array of non-negative ints.
+
+    For full documentation refer to :obj:`numpy.bincount`.
+
+    Parameters
+    ----------
+    x : {dpnp.ndarray, usm_ndarray}, 1 dimension, nonnegative ints
+        Input array.
+    weights : {dpnp.ndarray, usm_ndarray}, optional
+        Weights, array of the same shape as `x`.
+    minlength : int, optional
+        A minimum number of bins for the output array.
+
+    Returns
+    -------
+    out : dpnp.ndarray of ints
+        The result of binning the input array.
+        The length of `out` is equal to ``np.amax(x)+1``.
+
+    See Also
+    --------
+    dpnp.histogram, dpnp.digitize, dpnp.unique
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> np.bincount(np.arange(5))
+    array([1, 1, 1, 1, 1])
+    >>> np.bincount(np.array([0, 1, 1, 3, 2, 1, 7]))
+    array([1, 3, 1, 1, 0, 0, 0, 1])
+
+    >>> x = np.array([0, 1, 1, 3, 2, 1, 7, 23])
+    >>> np.bincount(x).size == np.amax(x)+1
+    True
+
+    The input array needs to be of integer dtype, otherwise a
+    TypeError is raised:
+
+    >>> np.bincount(np.arange(5, dtype=float))
+    Traceback (most recent call last):
+      ...
+    TypeError: Cannot cast array data from dtype('float64') to dtype('int64')
+    according to the rule 'safe'
+
+    A possible use of ``bincount`` is to perform sums over
+    variable-size chunks of an array, using the ``weights`` keyword.
+
+    >>> w = np.array([0.3, 0.5, 0.2, 0.7, 1., -0.6]) # weights
+    >>> x = np.array([0, 1, 1, 2, 2, 2])
+    >>> np.bincount(x,  weights=w)
+    array([ 0.3,  0.7,  1.1])
+
+    """
+
+    _bincount_validate(x, weights, minlength)
+
+    x, weights, usm_type = _ravel_check_a_and_weights(x, weights)
```

Comment:
Do we need to remove type casting inside `_ravel_check_a_and_weights` from bool dtype to int8 for `x` array? Since otherwise we will cast twice: from bool dtype to int8; from int8 to int64.

Reply:
Removed. Also removed test for it.
Now we don't have warning when casting happens and I don't think we should add one.